src/MICore/Debugger.cs

```diff
@@ -505,6 +534,12 @@ private bool IsLocalGdb()
             }
         }
 
+        private bool IsRemoteGdb()
+        {
+            return this.MICommandFactory.Mode == MIMode.Gdb &&
+               this._launchOptions is PipeLaunchOptions;
```

Comment:
Instead of explicitly checking for Pipe I would suggest 'anything besides local' since we already support other remote transports (ex: TCP), and we will probably add more in the future.